kfs-core/src/main/java/edu/arizona/kfs/fp/document/validation/impl/DisbursementVoucherNonEmployeeTravelValidation.java

```diff
@@ -0,0 +1,214 @@
+			errors.putError(KFSPropertyConstants.DISB_VCHR_TRAVEL_FROM_STATE_CODE, KFSKeyConstants.ERROR_DV_TRAVEL_FROM_STATE);
+			isValid = false;
+		}
+
+		if (KFSConstants.COUNTRY_CODE_UNITED_STATES.equals(nonEmployeeTravel.getDisbVchrTravelToCountryCode()) && StringUtils.isBlank(nonEmployeeTravel.getDisbVchrTravelToStateCode())) {
+			errors.putError(KFSPropertyConstants.DISB_VCHR_TRAVEL_TO_STATE_CODE, KFSKeyConstants.ERROR_DV_TRAVEL_TO_STATE);
+			isValid = false;
+		}
+
+		if (!isValid) {
+			errors.removeFromErrorPath(KFSPropertyConstants.DV_NON_EMPLOYEE_TRAVEL);
+			errors.removeFromErrorPath(KFSPropertyConstants.DOCUMENT);
+			return false;
+		}
+
+		/* must fill in all required per diem fields if any field is filled in */
+		boolean perDiemSectionComplete = validatePerDiemSection(document, errors);
+
+		/*
+		 * must fill in all required personal vehicle fields if any field is
+		 * filled in
+		 */
+		boolean personalVehicleSectionComplete = validatePersonalVehicleSection(document, errors);
+
+		/*
+		 * make sure per diem fields have not changed since the per diem amount
+		 * calculation
+		 */
+		if (!perDiemSectionComplete) {
+			isValid = false;
+		}
+
+		// validate the tax amount
+		isValid &= validateTravelAmount(document);
+
+		/*
+		 * make sure mileage fields have not changed since the mileage amount
+		 * calculation
+		 */
+		if (personalVehicleSectionComplete) {
+			KualiDecimal currentCalcAmt = document.getDvNonEmployeeTravel().getDisbVchrMileageCalculatedAmt();
+			KualiDecimal currentActualAmt = document.getDvNonEmployeeTravel().getDisbVchrPersonalCarAmount();
+			if (ObjectUtils.isNotNull(currentCalcAmt) && ObjectUtils.isNotNull(currentActualAmt)) {
+				KualiDecimal calculatedMileageAmount = getDisbursementVoucherTravelService().calculateMileageAmount(document.getDvNonEmployeeTravel().getDvPersonalCarMileageAmount(), document.getDvNonEmployeeTravel().getDvPerdiemStartDttmStamp());
+				if (calculatedMileageAmount.compareTo(document.getDvNonEmployeeTravel().getDisbVchrMileageCalculatedAmt()) != 0) {
+					errors.putErrorWithoutFullErrorPath(KFSConstants.GENERAL_NONEMPLOYEE_TAB_ERRORS, KFSKeyConstants.ERROR_DV_MILEAGE_CALC_CHANGE);
+					isValid = false;
+				}
+
+				// determine if the rule is flagged off in the parm setting
+				boolean performTravelMileageLimitInd = parameterService.getParameterValueAsBoolean(DisbursementVoucherDocument.class, DisbursementVoucherConstants.NONEMPLOYEE_TRAVEL_ACTUAL_MILEAGE_LIMIT_PARM_NM);
+				if (performTravelMileageLimitInd) {
+					// if actual amount is greater than calculated amount
+					if (currentCalcAmt.subtract(currentActualAmt).isNegative()) {
+						errors.putError(KFSPropertyConstants.DV_PERSONAL_CAR_AMOUNT, KFSKeyConstants.ERROR_DV_ACTUAL_MILEAGE_TOO_HIGH);
+						isValid = false;
+					}
+				}
+			}
+		}
+		
+		if (document.getDvNonEmployeeTravel().getDvPerdiemStartDttmStamp() != null || document.getDvNonEmployeeTravel().getDvPerdiemEndDttmStamp() != null) {
```

Comment:
If we do not check if the dates fields are null, the page will throw a stack trace error when the user intends to submit the document when the Travel tab is required. The reason for failing when null is because the statement inside is comparing if the dates were entered incorrectly
Null validation allows the document to state that the Dates fields are required without failing.